### PR TITLE
Save auth tokens to the OS keyring

### PR DESCRIPTION
**Pull request by Piccirello**
This prevents tokens from being written to the filesystem. Supported on macOS, Windows, and Linux.

### CODE REVIEW

File: pkg/configuration/config.go [hunk 1]
```diff
@@ -266,12 +295,44 @@ func Set(scope string, options map[string]string) {
...
-		config := configContents.Scoped[normalizedScope]
+		if key == models.ConfigToken.String() {
+			utils.LogDebug(fmt.Sprintf("Saving %s to system keyring", key))
+			uuid, err := uuid.NewRandom()
+			if err != nil {
+				utils.HandleError(err, "Unable to generate random UUID")
+			}
+			id := controllers.GenerateKeyringID(uuid.String())
+
+			if controllerError := controllers.SetKeyring(id, value); !controllerError.IsNil() {
```
Comment (rgmvisser) [inline pkg/configuration/config.go:312]: Can you add some comment to the different path here? It looks like this is some backward compatibility logic, but not sure

File: pkg/configuration/config.go [hunk 2]
```diff
@@ -266,12 +295,44 @@ func Set(scope string, options map[string]string) {
...
+				err := controllerError.Unwrap()
+				if err == keyring.ErrUnsupportedPlatform {
+					utils.LogDebug("Your OS does not support keyring")
+				} else {
+					utils.LogDebugError(err)
+					utils.LogDebug("Unable to access system keyring for secure storage")
+				}
```
Comment (rgmvisser) [inline pkg/configuration/config.go]:
```
utils.LogDebugError(controllerError.Unwrap())
utils.LogDebug(controllerError.Message)
```
?

**Review by rgmvisser — DISMISSED** [review]
Looks good in general, couple of clarification comments and nits.
Just to be sure, if the platform does not support keychain, for instance prod linux server, we just save it plain?

File: pkg/configuration/config.go (hunk 1 above)
Comment (Piccirello) [reply pkg/configuration/config.go:312]: Can you clarify what you mean here, I don't follow

Comment (rgmvisser) [reply pkg/configuration/config.go:312]: Nvm, ignore

**Comment by Piccirello** [comment]
>  Just to be sure, if the platform does not support keychain, for instance prod linux server, we just save it plain?

We discussed this in person, but for posterity: exactly!

File: pkg/configuration/config.go (hunk 2 above)
Comment (Piccirello) [reply pkg/configuration/config.go]: Updated, good catch

**Review by rgmvisser — DISMISSED** [review]

**Comment by Piccirello** [comment]
@rgmvisser pushed up a tiny change to make the keyring prefix check less error prone.

**Review by rgmvisser — APPROVED** [review]

**Pull request merged**